File: Problems/No.605_Can_Place_Flowers/can_place_flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {

        // start computing
        if (n == 0) return true;
        else {

            for (int i=0; i<flowerbed.size(); i++) {
                
                if (!flowerbed[i] && 
                    (((i == 0) || !flowerbed[i-1]) && 
                    ((i == flowerbed.size()-1) || !flowerbed[i+1]))) {
                    
                    flowerbed[i] = 1;
                    n--;

                    if (n == 0) return true;

                }

            }

        }

        return false;

    }
};

File: Problems/No.605_Can_Place_Flowers/can_place_flowers.py
class Solution(object):
    def canPlaceFlowers(self, flowerbed, n):
        """
        :type flowerbed: List[int]
        :type n: int
        :rtype: bool
        """

        # start computing
        if not n:
            return True
        else:
            for i in range(len(flowerbed)):
                if (not flowerbed[i] and
                    ((i == 0) or not flowerbed[i-1]) and
                    ((i == len(flowerbed)-1) or not flowerbed[i+1])):
                    n -= 1
                    if (n == 0):
                        return True
                    flowerbed[i] = 1
            return False
